src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,46 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int diagonalLength = matrix.length;
+        int[] diagonal = new int[diagonalLength];
+        for (int i = 0; i < diagonalLength; i++) {
+            diagonal[i] = matrix[i][i];
+        }
+        return diagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int diagonalLength = matrix.length;
+        int[] counterDiagonal = new int[diagonalLength];
+        for (int i = 0, j = diagonalLength - 1; i < diagonalLength; i++, j--) {
+            counterDiagonal[i] = matrix[i][j];
+        }
+        return counterDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int matrixShape = matrix.length;
+        int matrixSize = matrix.length * matrix.length;
+        int min = matrix[0][0];
+
+        for (int i = 1; i < matrixSize; i++) {
+            if (matrix[i / matrixShape][i % matrixShape] < min) {
```

Comment:
It's good that you are trying to optimize your code using only one loop , but you making sacrifice out of readability of your code. Save it for codewars and leetcode tasks) but here let's use two loops

Reply:
Got it. Two loops solution was successfully added